ui-react/packages/atlasmap/src/UI/ConditionalExpressionInput.tsx

```diff
@@ -1,30 +1,106 @@
 import React, { FunctionComponent } from "react";
-import { ToolbarGroup, ToolbarItem, Button } from "@patternfly/react-core";
-import { css, StyleSheet } from "@patternfly/react-styles";
 import {
-  // ExpressionContent,
-  IExpressionContentProps,
-} from "./ExpressionContent";
+  ToolbarGroup,
+  ToolbarItem,
+  Button,
+  Tooltip,
+} from "@patternfly/react-core";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { ExpressionContent } from "./ExpressionContent";
 
 const styles = StyleSheet.create({
   toolbarItem: { flex: 1 },
 });
+export interface IConditionalExpressionInputProps {
+  executeFieldSearch: (searchFilter: string, isSource: boolean) => string[][];
+  mappingExpressionAddField: (
+    selectedField: string,
+    newTextNode: any,
+    atIndex: number,
+    isTrailer: boolean,
+  ) => void;
+  mappingExpressionClearText: (
+    nodeId?: string,
+    startOffset?: number,
+    endOffset?: number,
+  ) => any;
+  isMappingExpressionEmpty: boolean;
+  mappingExpressionInit: () => void;
+  mappingExpressionInsertText: (
+    str: string,
+    nodeId?: string | undefined,
+    offset?: number | undefined,
+  ) => void;
+  mappingExpressionObservable: () => any;
+  mappingExpressionRemoveField: (
+    tokenPosition?: string,
+    offset?: number,
+    removeNext?: boolean,
+  ) => void;
+  onConditionalMappingExpressionEnabled: () => boolean;
+  onToggleExpressionMode: () => void;
+  mappingExpression?: string;
+  trailerId: string;
+}
+
+export const ConditionalExpressionInput: FunctionComponent<IConditionalExpressionInputProps> = ({
+  executeFieldSearch,
+  mappingExpressionAddField,
+  mappingExpressionClearText,
+  isMappingExpressionEmpty,
+  mappingExpressionInit,
+  mappingExpressionInsertText,
+  mappingExpressionObservable,
+  mappingExpressionRemoveField,
+  mappingExpression,
+  onConditionalMappingExpressionEnabled,
+  onToggleExpressionMode,
+  trailerId,
+}) => {
+  function onToggle() {
+    onToggleExpressionMode();
+  }
 
-export const ConditionalExpressionInput: FunctionComponent<IExpressionContentProps> = () =>
-  // props,
-  {
-    return (
-      <ToolbarGroup className={css(styles.toolbarItem)} role={"form"}>
-        <ToolbarItem>
-          <Button variant={"plain"} tabIndex={-1}>
+  return (
+    <ToolbarGroup className={css(styles.toolbarItem)} role={"form"}>
+      <ToolbarItem>
+        <Tooltip
+          content={"Enable/ Disable conditional mapping expression."}
+          enableFlip={true}
+          entryDelay={1000}
+          position={"left"}
+        >
+          <Button
+            variant={"plain"}
+            aria-label="Enable/ Disable conditional mapping expression"
+            tabIndex={-1}
+            onClick={onToggle}
+            disabled={!onConditionalMappingExpressionEnabled()}
```

Comment:
From this component standpoint, a `boolean` is enough to know if the component needs to be disabled or not. If calling a function to get the boolean is required, it should be done on the caller side.

Reply:
The boolean that controls whether its disabled is part of the mapping.  It needs a function call - this is the correct way to do this.

Reply:
`onConditionalMappingExpressionEnabled` is being called twice at each render. One here, the second time [here](https://github.com/atlasmap/atlasmap/pull/1982/files/3b94234a515c59699149785e94a3e251e302f0c4#diff-e5c48eedbcf5dc5fab5b849bda1e04e4R88). 

Ultimately, [this code](https://github.com/atlasmap/atlasmap/blob/d7f50903a3535a6cb8dfea91c6e3a6dbe74e6219/ui-react/packages/atlasmap-core/src/services/mapping-management.service.ts#L569) is being called, which does nothing but returning the value of a boolean.

I stand my point, this doesn't have to be a function because. Even more so, since it returns a value that deeply nested into the state used by RxJS, it's completely opaque to React's lifecycle. A change on that boolean will not trigger a render on the React side and will lead to a stale UI and bugs. I think right now it works only because we are registered to a number of observables on the `AtlasmapProvider` so that we "catch all changes" so to speak.